### PR TITLE
{Compute} command test coverage

### PR DESCRIPTION
**Pull request by cxznmhdcxz**
**Description**
Add command test coverage and record live test.
Included commands:
```
vm reapply
vm redeploy
vm unmanaged-disk list
vm convert
image list
image delete
image update
snapshot grant-access
snapshot revoke-access
```

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

**Review by wangzelin007 — APPROVED** [review]

**Review by wangzelin007 — COMMENTED** [review]
@cxznmhdcxz @zhoxing-ms
It seems we can not test `snapshot grant-access`.
image

**Comment by zhoxing-ms** [comment]
> @cxznmhdcxz @zhoxing-ms It seems we can not test `snapshot grant-access`. image

Replied above

File: scripts/ci/credscan/CredScanSuppressions.json [hunk 1]
```diff
@@ -171,7 +171,8 @@
         "src\\azure-cli\\azure\\cli\\command_modules\\vm\\tests\\latest\\recordings\\test_vm_managed_disk.yaml",
         "src\\azure-cli\\azure\\cli\\command_modules\\vm\\tests\\latest\\recordings\\test_vm_update_with_workspace_linux.yaml",
         "src\\azure-cli\\azure\\cli\\command_modules\\vm\\tests\\latest\\recordings\\test_vm_update_with_workspace_windows.yaml",
-        "src\\azure-cli\\azure\\cli\\command_modules\\vm\\tests\\latest\\recordings\\test_vm_boot_diagnostics.yaml"
+        "src\\azure-cli\\azure\\cli\\command_modules\\vm\\tests\\latest\\recordings\\test_vm_boot_diagnostics.yaml",
+        "src\\azure-cli\\azure\\cli\\command_modules\\vm\\tests\\latest\\recordings\\test_snapshot_access.yaml"
       ],
       "_justification": "[VM] response body contains random value recognized as secret in outdated recoding files of vm may remove in the future, and one-off password"
```
Comment (zhoxing-ms) [inline scripts/ci/credscan/CredScanSuppressions.json:177]: `test_snapshot_access` is not a VM related test. Please create a new item to exclude snapshot related test

File: scripts/ci/credscan/CredScanSuppressions.json [hunk 2]
```diff
@@ -175,6 +175,12 @@
...
     },
+    {
+      "file": [
+        "src\\azure-cli\\azure\\cli\\command_modules\\vm\\tests\\latest\\recordings\\test_snapshot_access.yaml"
+      ],
+      "_justification": "[Snapshot] response body contains random value recognized as secret"
```
Comment (zhoxing-ms) [inline scripts/ci/credscan/CredScanSuppressions.json:182]:
```suggestion
      "_justification": "[Snapshot] response body contains the expired SAS token from the snapshot that will be removed after the test is completed"
```

**Comment by cxznmhdcxz** [comment]
Close this. Please see new PR [21085](https://github.com/Azure/azure-cli/pull/21085).

**Pull request closed without merging**